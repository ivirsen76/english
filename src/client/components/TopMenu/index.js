import React from 'react'
import PropTypes from 'prop-types'
import { Dropdown, Menu, Modal } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'
import LoginForm from 'components/LoginForm'

const TopMenu = ({ isLoggedIn, user }) =>
    <Menu inverted>
        <MenuLink to="/" onlyActiveOnIndex>
            Home
        </MenuLink>
        <MenuLink to="/features">Features</MenuLink>
        <Menu.Item>Card base</Menu.Item>
        <Dropdown item text="Helpers">
            <Dropdown.Menu>
                <Dropdown.Item>Матрица Simple</Dropdown.Item>
                <Dropdown.Item>Матрица Continuous</Dropdown.Item>
                <Dropdown.Item>Выделение слов</Dropdown.Item>
                <Dropdown.Item>Числа на слух</Dropdown.Item>
            </Dropdown.Menu>
        </Dropdown>
        {isLoggedIn
            ? <Menu.Menu position="right">
                  <MenuLink to="/user">Dashboard</MenuLink>
                  <Dropdown
                      item
                      trigger={
                          <span>
                              <i className="icon-user" /> {user.email}
                          </span>
                      }
                  >
                      <Dropdown.Menu>
                          <Dropdown.Item>Change password</Dropdown.Item>
                          <MenuLink to="/logout">Log out</MenuLink>
                      </Dropdown.Menu>
                  </Dropdown>
              </Menu.Menu>
            : <Menu.Menu position="right">
                  <Modal size="small" closeIcon trigger={<a className="item">Log in</a>}>
                      <Modal.Header>Log in</Modal.Header>
                      <Modal.Content>
                          <LoginForm />
                      </Modal.Content>
                  </Modal>
              </Menu.Menu>}
    </Menu>

TopMenu.propTypes = {
    isLoggedIn: PropTypes.bool,
    user: PropTypes.object,
}

export default TopMenu
