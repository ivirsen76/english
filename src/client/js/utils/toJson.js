import enzymeToJson from 'enzyme-to-json'
import _forEach from 'lodash.foreach'

const convert = obj => {
    if (obj.props) {
        const newProps = {}

        _forEach(obj.props, (prop, key) => {
            newProps[key] = typeof prop === 'object' ? {} : prop
        })

        obj.props = newProps
    }

    if (obj.children) {
        obj.children = obj.children.map(child => convert(child))
    }

    return obj
}

export default wrapper => {
    const json = enzymeToJson(wrapper)
    return convert(json)
}
