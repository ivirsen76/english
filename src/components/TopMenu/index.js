import React, { PropTypes } from 'react'
import { Dropdown, Menu, Modal } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'
import LoginForm from 'components/LoginForm'


const TopMenu = ({ isLoggedIn, user }) => (
    <Menu inverted>
        <MenuLink to="/" onlyActiveOnIndex>Главная</MenuLink>
        <MenuLink to="/features">Возможности</MenuLink>
        <Menu.Item>Базы слов</Menu.Item>
        <Dropdown item text="Помощники">
            <Dropdown.Menu>
                <Dropdown.Item>Матрица Simple</Dropdown.Item>
                <Dropdown.Item>Матрица Continuous</Dropdown.Item>
                <Dropdown.Item>Выделение слов</Dropdown.Item>
                <Dropdown.Item>Числа на слух</Dropdown.Item>
            </Dropdown.Menu>
        </Dropdown>
        {isLoggedIn ? (
            <Menu.Menu position="right">
                <MenuLink to="/user">Dashboard</MenuLink>
                <Dropdown item trigger={<span><i className="icon-user" /> {user.email}</span>}>
                    <Dropdown.Menu>
                        <Dropdown.Item>Сменить пароль</Dropdown.Item>
                        <MenuLink to="/logout">Выйти</MenuLink>
                    </Dropdown.Menu>
                </Dropdown>
            </Menu.Menu>
        ) : (
            <Menu.Menu position="right">
                <Modal
                    size="small"
                    closeIcon
                    trigger={<a className="item">Войти</a>}
                >
                    <Modal.Header>Войти</Modal.Header>
                    <Modal.Content>
                        <LoginForm />
                    </Modal.Content>
                </Modal>
            </Menu.Menu>
        )}
    </Menu>
)

TopMenu.propTypes = {
    isLoggedIn: PropTypes.bool,
    user: PropTypes.object,
}

export default TopMenu
