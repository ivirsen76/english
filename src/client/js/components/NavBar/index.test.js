import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'js/utils/toJson'
import TopMenu from './index'

describe('<TopMenu />', () => {
    it('Should render menu', () => {
        const wrapper = shallow(<TopMenu />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Should render menu for logged user', () => {
        const wrapper = shallow(<TopMenu isLoggedIn user={{ email: 'some@one.com' }} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
