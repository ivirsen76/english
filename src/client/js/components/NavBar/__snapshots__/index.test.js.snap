// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopMenu /> Should render menu 1`] = `
<Menu
  inverted={true}
>
  <withProps(NavLink)
    exact={true}
    to="/"
  >
    Home
  </withProps(NavLink)>
  <withProps(NavLink)
    to="/features"
  >
    Features
  </withProps(NavLink)>
  <MenuItem>
    Card base
  </MenuItem>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    item={true}
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    text="Helpers"
  >
    <DropdownMenu>
      <DropdownItem>
        Matrix Simple
      </DropdownItem>
      <DropdownItem>
        Matrix Continuous
      </DropdownItem>
      <DropdownItem>
        Numbers
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <MenuMenu
    position="right"
  >
    <Modal
      closeIcon={true}
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      size="small"
      trigger={Object {}}
    >
      <ModalHeader>
        Log in
      </ModalHeader>
      <ModalContent>
        <Connect(Component) />
      </ModalContent>
    </Modal>
  </MenuMenu>
</Menu>
`;

exports[`<TopMenu /> Should render menu for logged user 1`] = `
<Menu
  inverted={true}
>
  <withProps(NavLink)
    exact={true}
    to="/"
  >
    Home
  </withProps(NavLink)>
  <withProps(NavLink)
    to="/features"
  >
    Features
  </withProps(NavLink)>
  <MenuItem>
    Card base
  </MenuItem>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    item={true}
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    text="Helpers"
  >
    <DropdownMenu>
      <DropdownItem>
        Matrix Simple
      </DropdownItem>
      <DropdownItem>
        Matrix Continuous
      </DropdownItem>
      <DropdownItem>
        Numbers
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <MenuMenu
    position="right"
  >
    <withProps(NavLink)
      to="/user"
    >
      Dashboard
    </withProps(NavLink)>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      icon="dropdown"
      item={true}
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      selectOnBlur={true}
      trigger={Object {}}
    >
      <DropdownMenu>
        <DropdownItem>
          Change password
        </DropdownItem>
        <withProps(NavLink)
          to="/logout"
        >
          Log out
        </withProps(NavLink)>
      </DropdownMenu>
    </Dropdown>
  </MenuMenu>
</Menu>
`;
